compiler/shared/test/diff/mono.mls

```diff
@@ -0,0 +1,1259 @@
+//│ fun main$$1() =
+//│   new OneInt$1 (#10)  match {case obj: OneInt$1 => fac$OneInt$1(obj)}
+//│ class OneInt$1(a) {
+//│ }
+//│ class OneInt(a: Int) {
+//│   fun fac: () -> Int
+//│ }
+//│ Int
+//│ res
+//│     = 1
+
+//:mono
+//:e // FIXME: Mutable Parameters
+//trait AnyFoo {
+//}
+//class FooPlus(#a): AnyFoo {
+//  fun bar(b) = a + b
+//}
+//class FooMinus(#a): AnyFoo {
+//  fun bar(b) = a - b
+//}
+//fun f(x) = x.bar(42)
+//f(new FooPlus(1))
+//f(new FooMinus(2))
+
+:mono
+val any = -20
+fun f(x) = 
+  if x > any then 0
+  else g(x - 1)
+fun g(x) = 
+  if x > any then g(x - 1)
+  else f(x - 2)
+g(1)
+//│ Parsed:
+//│ TypingUnit(NuFunDef(Some(false), any, None, [], IntLit(-20)), NuFunDef(None, f, None, [], Lam(Tup(_: Var(x)), Blk(...))), NuFunDef(None, g, None, [], Lam(Tup(_: Var(x)), Blk(...))), App(Var(g), Tup(_: IntLit(1))))
+//│ Lifted:
+//│ TypingUnit {
+//│   let any$3 = () => -20
+//│   fun f$1 = (x,) => {if (>(x, any,)) then 0 else g$2(-(x, 1,),)}
+//│   fun g$2 = (x,) => {if (>(x, any,)) then g$2(-(x, 1,),) else f$1(-(x, 2,),)}
+//│   Code(List(g$2(1,)))
+//│ }
+//│ Mono:
+//│ 
+//│ Defunc result: 
+//│ main$$3()
+//│ fun any$3() =
+//│   #-20
+//│ fun f$1(x) =
+//│   if >(x, any) then #0 else g$2(-(x, #1))
+//│ fun g$2(x) =
+//│   if >(x, any) then g$2(-(x, #1)) else f$1(-(x, #2))
+//│ fun main$$3() =
+//│   g$2(#1)
+//│ val any: -20
+//│ fun f: Int -> 0
+//│ fun g: Int -> 0
+//│ 0
+//│ any
+//│     = -20
+//│ res
+//│ Runtime error:
+//│   RangeError: Maximum call stack size exceeded
+
+:mono
+class OneInt(a: Int){
+  fun get = () -> a
+}
+class OneBool(b: Bool){
+  fun get = () -> b
+}
+(if b then new OneInt(1) else new OneBool(true)).get()
+//│ Parsed:
+//│ TypingUnit(NuTypeDef(class, OneInt, (), Tup(a: Var(Int)), (), None, None, TypingUnit(NuFunDef(None, get, None, [], Lam(Tup(), Var(a))))), NuTypeDef(class, OneBool, (), Tup(b: Var(Bool)), (), None, None, TypingUnit(NuFunDef(None, get, None, [], Lam(Tup(), Var(b))))), App(Sel(Bra(rcd = false, If(IfThen(Var(b), New(Some((TypeName(OneInt),[1,])), TypingUnit()), Some(New(Some((TypeName(OneBool),[true,])), TypingUnit())))), get), Tup()))
+//│ Lifted:
+//│ TypingUnit {
+//│   class OneInt$1([a: Int,]) {fun get = () => (this).a}
+//│   class OneBool$2([b: Bool,]) {fun get = () => (this).b}
+//│   Code(List(('(' if (b) then new OneInt$1([1,]) {} else new OneBool$2([true,]) {} ')').get()))
+//│ }
+//│ Mono:
+//│ 
+//│ Defunc result: 
+//│ main$$2()
+//│ fun get$OneInt$1(this) =
+//│   this.a
+//│ fun get$OneBool$2(this) =
+//│   this.b
+//│ fun main$$2() =
+//│   if b then new OneInt$1 (#1)  else new OneBool$2 (true)  match {case obj: OneInt$1 => get$OneInt$1(obj); case obj: OneBool$2 => get$OneBool$2(obj)}
+//│ class OneInt$1(a) {
+//│ }
+//│ class OneBool$2(b) {
+//│ }
+//│ class OneInt(a: Int) {
+//│   fun get: () -> Int
+//│ }
+//│ class OneBool(b: Bool) {
+//│   fun get: () -> Bool
+//│ }
+//│ Int | false | true
+//│ res
+//│     = true
+
+:mono
+:e // FIXME: Malformed Test?
```

Comment:
@yuankeyu could you comment on this and propose a fixa? What were you trying to do?